src/workerd/server/server.c++

```diff
@@ -2612,29 +2613,35 @@ kj::Promise<void> Server::listenOnSockets(config::Config::Reader config,
     continue;
 
   validSocket:
-    kj::Promise<kj::Own<kj::ConnectionReceiver>> listener = nullptr;
+    using PromisedReceived = kj::Promise<kj::Own<kj::ConnectionReceiver>>;
+    PromisedReceived listener = nullptr;
     KJ_IF_MAYBE(l, listenerOverride) {
       listener = kj::mv(*l);
     } else {
-      listener = network.parseAddress(addrStr, defaultPort)
-          .then([](kj::Own<kj::NetworkAddress> parsed) {
-        return parsed->listen();
-      });
+      listener = ([](kj::Promise<kj::Own<kj::NetworkAddress>> promise) -> PromisedReceived {
+        auto parsed = co_await promise;
+        co_return parsed->listen();
+      })(network.parseAddress(addrStr, defaultPort));
```

Comment:
I guess the `goto` means that you cannot simply turn `listenOnSockets` into a coroutine...

We should refactor this (later).